abaco.go

```diff
@@ -129,26 +129,32 @@ func (group *AbacoGroup) updateFrameTiming(p *packets.Packet, frameIdx FrameInde
 	if ts == nil {
 		return
 	}
+	newSubframeCount := frameIdx * abacoSubsampleDivisions
+	deltaSubframe := newSubframeCount - group.LastSubframeCount
 	deltaTs := ts.T - group.LastTimestamp.T
-	if frameIdx <= group.LastRowCount || deltaTs == 0 {
+	// There will be transient nonsense if the timestamp.Rate changes. I think the best
+	// approach is set subframe rate to 0. After the next (consistent) timestamp, it will recover.
+	unequalRates := ts.Rate != group.LastTimestamp.Rate
+	group.LastTimestamp = *ts
+	group.LastSubframeCount = newSubframeCount
+	if deltaSubframe <= 0 || deltaTs == 0 || unequalRates {
+		group.SubframeRate = 0
 		return
 	}
-	newRowCount := frameIdx * abacoPseudoRowRate
-	deltaRow := newRowCount - group.LastRowCount
-	group.CountsPerRow = deltaTs / uint64(deltaRow)
-	group.LastTimestamp = *ts
-	group.LastRowCount = newRowCount
+	group.SubframeRate = deltaTs / uint64(deltaSubframe)
```

Comment:
What? Why would a computer timestamp be involved in calculating the `SubframeRate`?

Reply:
Or should `Timestamp` be called `Framestamp`?

Reply:
Well, Mazsi's label was "timestamp", but it's closer to a "firmware base clock counter". It counts in 4 ns increments (at least, the current one does).

Reply:
Lets give it a name like `firmware_base_clock_counter` then. at least `firmware_timestamp` something to make very clear what it is, and that it is not `time.time()`